pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -288,19 +281,27 @@ func (c *awsCloudwatchScaler) GetMetrics(ctx context.Context, metricName string,
 
 	metric := external_metrics.ExternalMetricValue{
 		MetricName: metricName,
-		Value:      *resource.NewQuantity(int64(metricValue), resource.DecimalSI),
+		Value:      *resource.NewQuantity(metricValue*100, resource.DecimalSI),
```

Comment:
Why are you doing this? Multiplying this value * 100 changes current behaviour, doesn't?
I'd say that using an expression, this multiplier can be applied inside the expression, WDYT?

Reply:
@JorTurFer When debugging the scaler I've noticed that values are returned as whole numbers from the CloudWatch lib and are matching the value that I get when running the query directly in CloudWatch UI. After passing the metric to HPA, k8s stats shows that the value is not interpreted correctly and indeed auto-scale wasn't scaling up as expected, but multiplying it by 100 solved the issue. I haven't tested this scaler with a query builder, but I find it hard to believe that HPA would somehow work fine with the same value, just because it was fetched with a query builder in oppose to using expression.

Reply:
I get your point about the scaler behaviour because it's a fact that casting float64 into int64, we are losing information and multiplying it by 100, the information missed is less, but there are use cases where that situation is not a problem.
Apart from that, IMO we should maintain the value exposed by the upstream as it is given. This new feature gives to the users the option of multiplying the value as part of the expression, doesn't?

Reply:
I had to multiply by 100 even when the scaler received an integer value from CloudWatch (e.g. `values: ['12345']`, something like that. My query that I'm testing with is actually getting the max value, so it returns integers.

Reply:
Do you mean that you are getting `123,45` when you should receive `12345`?

Reply:
With this code:
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/4444781/163950357-3d374276-e167-4c3a-b8c8-a695226e3c73.png">

And with the following expression:
`SELECT MIN(MessageCount) FROM "AWS/AmazonMQ" WHERE Broker = 'test' and Queue = 'worker'`

It will print to log:
```
{"level":"debug","ts":1650353554.154571,"logger":"aws_cloudwatch_scaler","msg":"Received Metric Data","data":"{\n  MetricDataResults: [{\n      Id: \"q1\",\n      Label: \"MessageCount\",\n      StatusCode: \"Complete\",\n      Timestamps: [2022-04-19 07:31:00 +0000 UTC],\n      Values: [448]\n    }]\n}"}
{"level":"debug","ts":1650353554.154722,"logger":"aws_cloudwatch_scaler","msg":"Metric Data (float)","float":448}
{"level":"debug","ts":1650353554.1547341,"logger":"aws_cloudwatch_scaler","msg":"Metric Data (int)","int":448}
```

So it seems to me that the value is fetched from CloudWatch and converted from float to int without an issue and that the issue only starts when trying to load the metric into HPA. Am I missing something here?

Reply:
Do you know that the value displayed in the HPA is the calculated value (`metricValue/podCount`)? This is because as default the metric type is average (another contributor added support for target value weeks ago, but the default value is average).
I ask because a lot of people don't notice it, and maybe it could be the mismatch between the metrics.
If you want, you can [query the metric manually](https://keda.sh/docs/latest/operate/metrics-server/) just to ensure that KEDA is returning the proper value

Reply:
I haven't noticed that HPA displayed the `metricValue / podCount`. I'm displaying the HPA metrics with the [Lens app](https://k8slens.dev/). The number the shows there (after metric was multiplied by 100) matches the number that I see when I run the expression query directly inside AWS CloudWatch. Scaling up also doesn't work properly without multiplying the metric by 100. I run various deployments and none of them is set to 100 pods in the ScalingObject.

Reply:
Could you try [querying manually](https://keda.sh/docs/latest/operate/metrics-server/)? just to check if KEDA returns the correct value or not. If manually it works, the problem is outside KEDA, if not, it's inside

Reply:
OK, I've tried manual query with the version that multiplies the metric by 100 and got this weird result:
```
{
    "metricName": "s0-aws-cloudwatch-MessageCount",
    "metricLabels": null,
    "timestamp": "2022-04-21T08:00:45Z",
    "value": "4100"
  }
```

While this is what I see in the HPA stats inside the Lens app:
<img width="73" alt="image" src="https://user-images.githubusercontent.com/4444781/164408232-05dc246f-74e0-4daa-8653-1ef4492d9fb4.png">

Reply:
what is weird? 
That's the expected behavior, KEDA returns 100 and due to you have 50 instances, 100/50=2. 2 is the value that I can see in your lens picture. What is the value that you see in CloudWatch?

Reply:
OK. I didn't get the ClodWatch metric from their UI. I'll post another example with this soon.

Reply:
Here's an example with the version that multiplies by 100.

HPA stats in the Lens app:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/4444781/164410918-7d5c84e9-0b9a-48da-b0f4-652fbc8425ad.png">

Keda manual query result:
```
{
    "metricName": "s0-aws-cloudwatch-MessageCount",
    "metricLabels": null,
    "timestamp": "2022-04-21T08:15:10Z",
    "value": "4k"
  }
```

CloudWatch UI result:
<img width="241" alt="image" src="https://user-images.githubusercontent.com/4444781/164411042-2774478c-be3c-4897-b1a1-f994410c7921.png">

Reply:
yes, and that's what I tried to explain. The value that you can see in the HPA is the result of the formula (returnedValue/podCount). You cannot compare it with the value from the upstream directly, you need to apply the same formula. 
- CloudWatch -> 40
- KEDA -> 4k (40 * 100)
- HPA -> 80 (4000/50)
This is wrong because the expected values should be:
- CloudWatch -> 40
- KEDA -> 40
- HPA -> 800m (40/50) *Remember that in k8s, decimal numbers don't exist, 0.8 is translated into mili scale so 0.8 -> 800m

You shouldn't multiply the upstream value * 100 because the behavior changes. 
When I said to multiply inside the expression, I though that your case was related with working with rates between 0 and 1 and you need to do it because the information lost during the casting to int64.

Reply:
I see, so what is `targetAverageValue` actually referring to? How can I set it up correctly?

Reply:
`targetAverageValue` is the desired value per pod. Imagine that your workload can process one message at once and you want to process all messages faster as possible, in that case, you could set `targetAverageValue` to 1, to have 1 pod per message.
In the same case, if your pod can process 2 messages at once (and you want to process them faster as possible with less resources as possible), the `targetAverageValue` value will be 2.

if 10 messages are pending , with `targetAverageValue` 1 you will have 10 pods (10/1=10) and with 2 you will have 5 pods (10/2=5)

No worries about the spent time, I'm happy to help 😄 

Let me know if you have any doubt about how it works

Reply:
Awesome, thanks. Now it make sense. I hope to get the tests & docs changes in the next few days. Have a great day!

Reply:
sure, no rush at all
Have a great day too